java-checks/src/main/java/org/sonar/java/checks/CallSuperInTestCaseCheck.java

```diff
@@ -0,0 +1,121 @@
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+
+@Rule(
+  key = "S2188",
+  name = "JUnit test cases should call super methods",
+  tags = {"bug", "junit"},
+  priority = Priority.CRITICAL)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.UNIT_TESTABILITY)
+@SqaleConstantRemediation("5min")
+@ActivatedByDefault
+public class CallSuperInTestCaseCheck extends SubscriptionBaseVisitor {
+
+  public static final String JUNIT_FRAMEWORK_TEST_CASE = "junit.framework.TestCase";
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.METHOD);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    MethodTree methodTree = (MethodTree) tree;
+    Symbol.MethodSymbol methodSymbol = methodTree.symbol();
+    boolean isMethodInJunit3 = isWithinJunit3TestCase(methodSymbol) && isSetUpOrTearDown(methodSymbol);
+    if(isMethodInJunit3 && requiresSuperCall(methodSymbol) && !callSuperOnOverride(methodTree.block(), methodSymbol)) {
+      addIssue(tree, String.format("Add a \"super.%s()\" call to this method.", methodSymbol.name()));
+    }
+  }
+
+  private boolean requiresSuperCall(Symbol.MethodSymbol methodSymbol) {
+    Type superType = methodSymbol.owner().type().symbol().superClass();
+    Collection<Symbol> symbols = Lists.newArrayList();
+    while(!superType.is(JUNIT_FRAMEWORK_TEST_CASE) && symbols.isEmpty()) {
+     symbols = superType.symbol().lookupSymbols(methodSymbol.name());
+     superType = superType.symbol().superClass();
+    }
+    return !symbols.isEmpty() && !symbols.iterator().next().owner().type().is(JUNIT_FRAMEWORK_TEST_CASE);
+  }
+
+  private boolean callSuperOnOverride(@Nullable BlockTree block, Symbol.MethodSymbol methodSymbol) {
+    if(block == null) {
+      return false;
+    }
+    InvocationVisitor visitor = new InvocationVisitor(methodSymbol.name());
+    block.accept(visitor);
+    return visitor.superCallOnOverride;
+  }
+
+  private boolean isWithinJunit3TestCase(Symbol.MethodSymbol methodSymbol) {
+    Type type = methodSymbol.owner().type();
+    return type.isSubtypeOf(JUNIT_FRAMEWORK_TEST_CASE) && !type.symbol().superClass().is(JUNIT_FRAMEWORK_TEST_CASE);
+  }
+
+  private boolean isSetUpOrTearDown(Symbol.MethodSymbol methodSymbol) {
+    return (methodSymbol.name().equals("setUp") || methodSymbol.name().equals("tearDown"))
+        && methodSymbol.parameterTypes().size() == 0 && methodSymbol.returnType().type().is("void");
```

Comment:
You don't need to test the return type as with the same name and number of parameters you are already sure of the override.